prqlc/prqlc/src/semantic/lowering.rs

```diff
@@ -19,15 +19,22 @@ use crate::COMPILER_VERSION;
 use crate::{Error, Reason, Span, WithErrorInfo};
 use prqlc_ast::expr::generic::{InterpolateItem, Range, SwitchCase};
 
-use super::NS_DEFAULT_DB;
-
-/// Convert AST into IR and make sure that:
+/// Convert a resolved expression at path `main_path` relative to `root_mod`
+/// into RQ and make sure that:
 /// - transforms are not nested,
 /// - transforms have correct partition, window and sort set,
 /// - make sure there are no unresolved expressions.
+///
+/// All table references must reside within module at `database_module_path`.
+/// They are compiled to table identifiers, using their path relative to the database module.
+/// For example, with `database_module_path=my_database`:
+/// - `my_database.my_table` will compile to `"my_table"`,
+/// - `my_database.my_schema.my_table` will compile to `"my_schema.my_table"`,
+/// - `my_table` will error out saying that this table does not reside in current database.
```

Comment:
What's the utility of this? Relative to leaving them as they're specified in the query? (not claiming confidently that there isn't an advantage!)

Reply:
or this will obviate the need for this?

> Fully qualified name is used to refer to that table. This will be changed soon.

Reply:
This adds an ability to specify a module as a "database module":

```elm
module my_database {
    module my_schema {
        let my_table <[{}]>   # this will be `my_schema.my_table`
    }
}

module another_database {
    let another_table <[{}]>  # this will be `another_database`
}
```
The problem is that these two tables are defined in two different databases, so they cannot be present in the same query. That's why "current database" is a compiler parameter, so it knows for which database it is compiling for.

Idea being that in the future, we could support querying multiple databases, so PRQL source would compile into multiple queries that would be sent to each of the databases.